### PR TITLE
Fixed the place having duplicates

### PR DESCRIPTION
**Pull request by sima-zhu**
Java toc currently get one package from `Get-Unique`
Fixed the problem on checking all columns. Since the packages with same service name are not that many, the call does not take that many time.

Before fix:
core service only returns 'azure-core'
After fix:
core service returns all core packages, e.g. 'azure-core-okhttp', 'azure-core-test' etc

### CODE REVIEW

File: eng/common/scripts/Update-DocsMsToc.ps1 [hunk 1]
```diff
@@ -90,17 +90,6 @@ function GetPackageLookup($packageList) {
         Write-Host "Found more than one package entry for $($pkg.Package) selecting the first non-hidden one."
       }
     }
-
```
Comment (sima-zhu) [inline eng/common/scripts/Update-DocsMsToc.ps1:93]: pkgKey has already taken this into account.

File: eng/common/scripts/Update-DocsMsToc.ps1 [hunk 2]
```diff
@@ -177,23 +167,26 @@ foreach ($package in $packagesForToc.Values) {
...
   $packageItems = @()
 
   # Client packages get individual entries
   $clientPackages = $packagesForToc.Values.Where({ $_.ServiceName -eq $service -and ('client' -eq $_.Type) })
-  $clientPackages = $clientPackages | Sort-Object 'Package', 'GroupId' | Get-Unique
-  if ($clientPackages) {
-    foreach ($clientPackage in $clientPackages) {
+  $clientPackages = $clientPackages | Select-Object * -Unique
+  foreach ($clientPackage in $clientPackages) {
+    if ($service -eq 'Core') {
```
Comment (weshaggard) [inline eng/common/scripts/Update-DocsMsToc.ps1:182]: As we talked earlier this is likely to change to "Other - Core" so we will need to figure out. Why do we need to separate the core packages out directly?

Comment (sima-zhu) [reply eng/common/scripts/Update-DocsMsToc.ps1:182]: I thought JS has already made this change (put core under other) and we have to follow what we have. Is this still in discussion?
@danieljurek do you know the current status? Where is the best place for 'core', parallel with other services or put under 'Other'?

Comment (sima-zhu) [reply eng/common/scripts/Update-DocsMsToc.ps1:182]: Oh, I know what you mean. Do you mean JS is able to have core under other, why we need to literally do this for Java? I am going to check why JS is able to do this but Java cannot.

Comment (sima-zhu) [reply eng/common/scripts/Update-DocsMsToc.ps1:182]: In JS csv, the service name for core is 'Other'. The right path is to change over there. Will remove the logic here

Comment (danieljurek) [reply eng/common/scripts/Update-DocsMsToc.ps1]: Nesting in the `Other` service is controlled by dashes: 

If you want: 
* Other
  * Core
    * Client
      * AMQP 
      * HTTP Netty
      * etc...

Then set the service name to "Other" and the display name to (for AMQP) 

```
Core - Client - AMQP
```

File: eng/common/scripts/Update-DocsMsToc.ps1 [hunk 3]
```diff
@@ -116,7 +105,8 @@ $onboardedPackages = &$GetOnboardedDocsMsPackagesFn `
 $metadata = (Get-CSVMetadata).Where({
     $_.Package `
       -and $onboardedPackages.ContainsKey($_.Package) `
-      -and $_.Hide -ne 'true'
+      -and $_.Hide -ne 'true' `
+      -and $_.New -eq 'true'
```
Comment (danieljurek) [inline eng/common/scripts/Update-DocsMsToc.ps1:109]: We shouldn't filter by `$_.New -eq 'true'` here. 

We're not generating the ToC based entirely on what's in metadata. We start with the list of packages that are onboarded and then we add metadata information to those onboarded packages... Whether or not a package is `New` doesn't affect if a package is already onboarded in the docs (usually under the `Legacy` moniker)

**Review by weshaggard — APPROVED** [review]
The change looks OK to me but I would like @danieljurek to also have a look.

**Review by danieljurek — COMMENTED** [review]
Let's move this to the next step so we can get PRs opened and tested for JS to verify the chainges in `main`

**Pull request merged**